File: backend/app/schemas/billing.py
"""Billing-related Pydantic schemas."""

from datetime import datetime
from typing import Literal, Optional
from pydantic import BaseModel


class CreateCheckoutSessionRequest(BaseModel):
    """Request to create a Stripe checkout session."""
    price_type: Literal["monthly", "annual"]


class CheckoutSessionResponse(BaseModel):
    """Response with checkout URL."""
    checkout_url: str


class PortalSessionResponse(BaseModel):
    """Response with customer portal URL."""
    portal_url: str


class SubscriptionStatus(BaseModel):
    """Current subscription status."""
    plan_type: str
    max_users: int
    max_projects: Optional[int]
    current_users: int
    current_projects: int
    subscription_status: Optional[str]
    current_period_end: Optional[datetime]
    cancel_at_period_end: bool = False


class WebhookEvent(BaseModel):
    """Stripe webhook event wrapper."""
    type: str
    data: dict

File: frontend/src/api/billing.ts
import { client } from './client';

export interface SubscriptionStatus {
  plan_type: string;
  max_users: number;
  max_projects: number | null;
  current_users: number;
  current_projects: number;
  subscription_status: string | null;
  current_period_end: string | null;
  cancel_at_period_end: boolean;
}

export const createCheckoutSession = (plan: 'starter' | 'team', priceType: 'monthly' | 'annual') =>
  client.post<{ checkout_url: string }>('/billing/create-checkout-session', { plan, price_type: priceType });

export const createPortalSession = () =>
  client.post<{ portal_url: string }>('/billing/create-portal-session');

export const getSubscription = () =>
  client.get<SubscriptionStatus>('/billing/subscription');
